Sources/DolbyIORTSCore/State/StateMachine.swift

```diff
@@ -110,32 +89,15 @@ final actor StateMachine {
     }
 
     func onSubscribed() {
-        switch currentState {
-        case .subscribing:
-            currentState = .subscribed(.init())
-        default:
-            Self.logger.error("🛑 Unexpected state on onSubscribed - \(self.currentState.description)")
-        }
+        currentState = .subscribed(.init())
```

Comment:
Have you got a chance to test the behaviour with and without the auto connect option @ldanilina  ?

Reply:
Yes, I think I've finished with it.. seems like the issue is gone with `options.autoReconnect = false`, updated PR